code/EventLinkQR/app/src/main/java/com/example/eventlinkqr/CreateEventFragment.java

```diff
@@ -197,15 +208,73 @@ public int getCount() {
                     // delete the previous poster if the user removed the poster
                     if(image == null){
                         ImageManager.deletePoster(newEvent.getId());
-                    }else{
+                    } else {
                         ImageManager.uploadPoster(getContext(), newEvent.getId(), image);
                     }
-                }else{
-                    // if the user, hasn't set a limit, set it to a default value
-                    if(maxAttendee.equals("")) {
-                        EventManager.createEvent(getContext(), newEvent, organizer, customQRString, Integer.MAX_VALUE, image);
-                    }else{
-                        EventManager.createEvent(getContext(), newEvent, organizer, customQRString, Integer.parseInt(maxAttendee), image);
+                } else {
+                    // Check if the QR code is already in use
+                    if (customQRString != null && customQRString.equals(customPromotionalQRString)) {
+                        Toast.makeText(requireActivity(), "Check in and promotional code cannot be the same", Toast.LENGTH_LONG).show();
+                    } else if (customQRString != null && customPromotionalQRString != null) {
+                        Bitmap finalImage = image;
+                        QRCodeManager.fetchQRCode(customQRString).addOnSuccessListener(code ->{
+                            try {
+                                checkQRCodeActive(code);
+                            } catch (QRCodeAlreadyActiveException e) {
+                                Toast.makeText(requireActivity(), "Check in code already in use", Toast.LENGTH_LONG).show();
+                                return;
+                            }
+
+                            QRCodeManager.fetchQRCode(customPromotionalQRString).addOnSuccessListener(pCode -> {
+                                try {
+                                    checkQRCodeActive(pCode);
+                                } catch (QRCodeAlreadyActiveException e) {
+                                    Toast.makeText(requireActivity(), "Promotional code already in use", Toast.LENGTH_LONG).show();
+                                    return;
+                                }
+
+                                createEvent(newEvent, organizer, maxAttendee, finalImage);
```

Comment:
The maxAttendees is not getting checked here in case it's blank or in any of the if statements other than the last. This will cause an error when the field is blank.

Reply:
I will add a comment to make this clearer, but `createEvent` here is a private helper method that then calls `EventManager.createEvent`. This helper checks `maxAttendee` and converts from string to int.

Reply:
Actually, I just renamed the method and added javadoc to it